schemas/v1.0/azure.yaml.json

```diff
@@ -211,7 +211,7 @@
                             "properties": {
                                 "dist": {
                                     "type": "string",
-                                    "description": "The CLI will use the JAR file in this directory to create the deployment artifact (ZIP file). If omitted, the CLI will detect the output directory based on the build system in-use."
+                                    "description": "Optional. The path to the directory containing a single Java archive file (.jar/.ear/.war), or the path to the specific Java archive file to be included in the deployment artifact. If omitted, a smart default is chosen based on the configured language."
```

Comment:
> If omitted, a smart default is chosen based on the configured language.

I think that this description is specific to the case where you have set the language to be "java" (if the JSON Schema interpreter in my head is correct).  I wonder if we should change this to be a little more specific about the actual heuristic (i.e. if there is a single .jar/.war/.ear file, file we use it, otherwise you have to specify?)

Reply:
@ellismg May need a little help, I think I've tried to make the current wording to reflect that but I can see how that is easily missed (the keyword *single*):

> Optional. The path to the directory containing a single Java archive file (.jar/.ear/.war), or the path to the specific Java archive file to be included in the deployment artifact. If omitted, a smart default is chosen based on the configured language

Reply:
I guess I struggle a little with the "is chosen based on the configured language" since we know in this case it is just "java", right? But what you have here seems good enough to ship.  Always easy to change the prose in our documentation later.

Reply:
Ah, I should not have changed that part. In my mind, I'm still worried about when we add `gradle` support. "build system" is more correct, and I'll revert it to the previous verbiage.

Reply:
Ahh - Yes - this all makes sense to me.  Your suggestion looks good!

Reply:
Ah. I should not have updated that part. In my mind, there's a world where `gradle` exists. I've appended a snippet:

> For maven, the default output directory 'target' is assumed.